### PR TITLE
Init pretty auth

### PR DESCRIPTION
**Pull request by Avaxapsys**

### CODE REVIEW

File: DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -1,94 +1,53 @@
...
             _userService = userService;
             _mapper = mapper;
         }
 
         [HttpPost("/register")]
+        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
         public int Register([FromBody] UserInsertInputModel model)
         {
+            if(model.Roles.Count == 0)
+                model.Roles.Add(Role.Student);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs:40]: this logic must be in BLL

File: DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Text;
+using Microsoft.IdentityModel.Tokens;
+
+namespace DevEdu.Business.Configuration
+{
+    public class AuthOptions
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs:6]: we have the same in API-project
so please remove this class from BLL

File: DevEdu.API/DevEdu.Business/Services/AuthenticationService.cs [hunk 3]
```diff
@@ -1,23 +1,47 @@
...
+using Microsoft.IdentityModel.Tokens;
 
 namespace DevEdu.Business.Services
 {
-    public class AuthenticationService
+    public class AuthenticationService : IAuthenticationService
     {
-        public byte[] GetSalt()
+        private readonly IUserService _userService;
+        public AuthenticationService(){}
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business/Services/AuthenticationService.cs:15]: default constructor must be removed

File: DevEdu.API/DevEdu.Business/Services/UserService.cs [hunk 4]
```diff
@@ -15,17 +15,16 @@ public UserService(IUserRepository userRepository)
 
         public int AddUser(UserDto dto)
         {
-           var addedUserId = _userRepository.AddUser(dto);
+            AuthenticationService authenticationService = new AuthenticationService();
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Business/Services/UserService.cs:18]: 2 problems are here:
1. each service must be injected if it is present in the DI-setup
2. it's very bad to inject one service into another. the better solution is to call authenticationService from the controller

File: DevEdu.API/DevEdu.Db/StoredProcedures/User/User_SelectByEmail.sql [hunk 5]
```diff
@@ -3,24 +3,24 @@
...
+		,U.IsDeleted
+		,U.RegistrationDate
+		,U.ContractNumber
+		,U.BirthDate
+		,U.GitHubAccount
+		,U.Photo
+		,U.PhoneNumber
+		,U.ExileDate
+		,U.CityId as id
+		,UR.RoleId as id
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.Db/StoredProcedures/User/User_SelectByEmail.sql:22]: please revert your changes
I'm going to share one more doc. from our partners, and it contains such words:
image

File: DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs (hunk 1 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs:40]: +

File: DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs (hunk 2 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs:6]: I use it in BLL, that has no dependency from API, should I delete it from API?

File: DevEdu.API/DevEdu.Business/Services/UserService.cs (hunk 4 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/Services/UserService.cs:18]: it's awesome

File: DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs (hunk 2 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/Configuration/AuthOptions.cs:6]: replaced to BLL

File: DevEdu.API/DevEdu.Business/Services/AuthenticationService.cs (hunk 3 above)
Comment (Avaxapsys) [reply DevEdu.API/DevEdu.Business/Services/AuthenticationService.cs:15]: I did it for using in userService, thanks for advice.

File: DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        [HttpPost("/register")]
+        [ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
+        public int Register([FromBody] UserInsertInputModel model)
+        {
+            var dto = _mapper.Map<UserDto>(model);
+            var addedUser = _userService.AddUser(dto, _authService);
+            return addedUser;
+        }
+
+        [HttpPost("/signin")]
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/AuthenticationController.cs:37]: sign-in

**Pull request merged**